data-prepper-plugins/s3-source/src/test/java/com/amazon/dataprepper/plugins/source/SqsQueueUrlTest.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.source;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import java.net.MalformedURLException;
+import java.util.UUID;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class SqsQueueUrlTest {
+    @Test
+    void parse_throws_when_URL_is_not_a_URL() {
+        assertThrows(MalformedURLException.class, () -> SqsQueueUrl.parse(UUID.randomUUID().toString()));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {
+            "https://sqs.us-east-1.amazonaws.com",
+            "https://sqs.us-east-1.amazonaws.com/",
+            "https://sqs.us-east-1.amazonaws.com/123456789",
+            "https://sqs.us-east-1.amazonaws.com/123456789/"
+    })
```

Comment:
The following should be invalid too: `https://sqs.us-east-2.amazonaws.com/123456789012/MyQueue`

Reply:
How is this an invalid URL?